content/r-package-dev/Documentation.Rmd

```diff
@@ -0,0 +1,81 @@
+---
+title: "Documentation"
+date: "9999-07-31"
+author: "Lindsay R. Carr"
+slug: "doc"
+image: "img/main/intro-icons-300px/r-logo.png"
+output: USGSmarkdowntemplates::hugoTraining
+parent: R Package Development
+weight: 1
+---
+
+```{r setup, include=FALSE, warning=FALSE, message=FALSE}
+library(knitr)
+
+knit_hooks$set(plot=function(x, options) {
+  sprintf("<img src='../%s%s-%d.%s'/ title='%s'/>", 
+          options$fig.path, options$label, options$fig.cur, options$fig.ext, options$fig.cap)
+
+})
+
+opts_chunk$set(
+  echo=TRUE,
+  fig.path="static/doc/",
+  fig.width = 6,
+  fig.height = 6,
+  fig.cap = "TODO"
+)
+
+set.seed(1)
+```
+
+The quality of documentation for an R package can determine it's success or failure. Less documentation generally results in higher burdens for package developer's, so it's best add extensive documentation as early as possible. Consider 3 different audiences when writing pacakge documentation: the users, current or future developers, and future you. The better documented your package the less like you are to be burdened by answering user questions, helping onboard future developers, and spending time remembering details about the code.
+
+## Lesson Objectives
+
+1. Distinguish the different types of documentation for R packages.
+2. Develop documentation for individual functions.
+3. Create a vignette to highlight the top-level package uses.
+4. Edit and update README files.
+
+## Components of package documentation
+
+There are multiple components of package documentation. 
+
+- **README:** In the README, you include directions for how to quickly get setup and started. It might include a simple workflow or call out specific functions that people might find interesting. 
+- **Function help files:** Help files are the files that document the inputs and outputs of a function, references and links to related functions, define defaults for the user, and give examples of how to use the function. These files apply to functions at an individual level and are created using Roxygen syntax (we will use the [`roxygen2` package](https://cran.r-project.org/web/packages/roxygen2/vignettes/roxygen2.html).
+- **Vignettes:** Packages can have multiple vignettes to illustrate potential workflows or topics within the package. They generally are high-level, illustrating how to apply the package functions to complete an analysis.
+
+The README was already discussed in [*What is a Package?*](/pkg-def), so we won't be covering it further here. Just remember that this is a useful way to give someone a brief introduction to your package.
```

Comment:
OK, so we can have that section mention READMEs and then point to this one to learn more :)